mongo/gridfs/bucket.go

```diff
@@ -469,12 +418,11 @@ func (b *Bucket) GetChunksCollection() *mongo.Collection {
 	return b.chunksColl
 }
 
-func (b *Bucket) openDownloadStream(filter interface{}, opts ...*options.FindOptions) (*DownloadStream, error) {
-	ctx, cancel := deadlineContext(b.readDeadline)
-	if cancel != nil {
-		defer cancel()
-	}
-
+func (b *Bucket) openDownloadStream(
+	ctx context.Context,
+	filter interface{},
+	opts ...*options.FindOptions,
+) (*DownloadStream, error) {
 	cursor, err := b.findFile(ctx, filter, opts...)
```

Comment:
Existing bug: This cursor must be closed. Could `findFile` use `FindOne` instead?